discord/ext/tasks/__init__.py

```diff
@@ -428,9 +468,55 @@ def error(self, coro):
         return coro
 
     def _get_next_sleep_time(self):
-        return self._last_iteration + datetime.timedelta(seconds=self._sleep)
+        if self._sleep is not None:
+            return self._last_iteration + datetime.timedelta(seconds=self._sleep)
+
+        if self._time_index >= len(self._time):
+            self._time_index = 0
+
+        next_time = self._time[self._time_index]
 
-    def change_interval(self, *, seconds=0, minutes=0, hours=0):
+        if self._current_loop == 0:
+            self._time_index += 1
+            return datetime.datetime.combine(datetime.datetime.now(datetime.timezone.utc), next_time)
+
+        next_date = self._last_iteration
+        if self._time_index == 0:
+            # we can assume that the earliest time should be scheduled for "tomorrow"
+            next_date += datetime.timedelta(days=1)
```

Comment:
So how does this assumption work if it's 6PM and we have the following times set?

```py
times=[
    time(19, 0),
    time(20, 0),
    time(22, 0),
]
```

Is this only fired on 7PM the next day?

Reply:
Good question! I actually caught and fixed this bug during my own testing.

When the loop is started, the first time is slept to: https://github.com/Rapptz/discord.py/pull/6456/files#diff-d612a21a8825c4e1754417b8840eed107ac96e41def4c0c92cf6394300921b51R94-R101 (lines 94-101)

In the case of an already running loop, the timedelta is added to the previous iteration time, so it will always be the "next day" relative to the previous time since `self._time` is sorted.